x-pack/plugins/security_solution/server/lib/sourcerer/routes/index.ts

```diff
@@ -26,85 +35,220 @@ export const createSourcererDataViewRoute = (
         body: buildRouteValidation(sourcererSchema),
       },
       options: {
+        authRequired: true,
         tags: ['access:securitySolution'],
       },
     },
     async (context, request, response) => {
       const siemResponse = buildSiemResponse(response);
       const siemClient = context.securitySolution?.getAppClient();
       const dataViewId = siemClient.getSourcererDataViewId();
+      const unsecuredSavedObjectClient = context.core.savedObjects.getClient({
+        excludedWrappers: ['security'],
+      });
       try {
         const [
           ,
           {
             data: { indexPatterns },
           },
         ] = await getStartServices();
-        const dataViewService = await indexPatterns.indexPatternsServiceFactory(
+        const auditLogger = security?.audit.asSystem(request);
+        /*
+         * Note for future engineer
+         * We need to have two different DataViewService because one will be to access all
+         * the data views that the user can access and the UnsecuredDataViewService will be
+         * to create the security solution data view
+         */
+        const unsecuredDataViewService = await indexPatterns.dataViewsServiceFactory(
+          unsecuredSavedObjectClient,
+          context.core.elasticsearch.client.asCurrentUser,
+          request,
+          true
+        );
+        const dataViewService = await indexPatterns.dataViewsServiceFactory(
           context.core.savedObjects.client,
-          context.core.elasticsearch.client.asInternalUser,
+          context.core.elasticsearch.client.asCurrentUser,
           request
         );
 
-        let allDataViews = await dataViewService.getIdsWithTitle();
+        let allDataViews: DataViewListItem[] = [];
+        try {
+          allDataViews = await dataViewService.getIdsWithTitle();
+        } catch (error) {
+          // DO nothing because we will return at least one data view.
+          // let's not block the security solution users
+        }
+        let siemDataView;
+        try {
+          siemDataView = await unsecuredDataViewService.get(dataViewId);
+          auditLogger?.log(
+            sourcererSavedObjectEvent({
+              action: SavedObjectAction.GET,
+              id: dataViewId,
+            })
+          );
+        } catch (error) {
+          // if does not exist, it is all good
```

Comment:
This could fail for an untold number of reasons. Is it really OK to ignore _all_ errors, or do you want to do something else in the event that it's not a "not found" error?